src/main/java/com/connexta/search/query/QueryManagerImpl.java

```diff
@@ -98,31 +130,19 @@ public QueryManagerImpl(
         .collect(Collectors.toList());
   }
 
-  /**
-   * Creates the {@link Filter} represented by the {@code queryString}
-   *
-   * @throws MalformedQueryException if the {@code queryString} cannot be parsed
-   * @throws IllegalQueryException if the {@code queryString} contains unsupported attributes
-   */
-  @VisibleForTesting
-  static Filter getFilter(final String queryString)
-      throws MalformedQueryException, IllegalQueryException {
-    final Filter filter;
-    try {
-      filter = CQL.toFilter(queryString);
-    } catch (final CQLException e) {
-      throw new MalformedQueryException(e);
-    }
-
-    final Set<String> unsupportedAttributes =
-        Filters.propertyNames(filter).stream()
-            .map(PropertyName::getPropertyName)
-            .filter(attribute -> !QUERY_TERMS.contains(attribute))
-            .collect(Collectors.toSet());
-    if (!unsupportedAttributes.isEmpty()) {
-      throw new IllegalQueryException(unsupportedAttributes);
+  private boolean isKnownError(Exception e) {
```

Comment:
The the method `isKnownError` is grotesque, but the code to ask Solr if there are any documents in the core is even worse:

```
   CoreAdminRequest req = new CoreAdminRequest();
   req.setAction(CoreAdminParams.CoreAdminAction.STATUS);
   req.setCoreName("searchTerms");
   NamedList<Object> response;
   response = new 
       HttpSolrClient.Builder().withBaseSolrUrl("http://solr:8983/solr/").build().request(req);
   return response.get("status").get("searchTerms").get("index").get("numDocs") > 0
```
And there is a try-catch block for a checked exception that is not shown in this sample code.

Reply:
add a TODO comment to replace this hack